### DISCUSSION TITLE
Confusion (and possibly PEBKAC) regarding price_limit

### DISCUSSION THREAD

**Discussion by kawlsson** [kotope/aio_energy_management#54] [Q&A]
Hi,
I've used this component for a few weeks and i really like it. It works perfectly so far.
However, i now wanted to add a price limit so i don't burn through the battery just because the price is a few fractions higher a few hours a day.
But when i add it it, it stops working and i cannot understand why. Any help would be appreciated.
My current config:
```
      - nordpool_entity: sensor.nordpool_kwh_se3_sek_3_10_0
        unique_id: nordpool_aio_expensive_hours
        name: Nordpool AIO Expensive Hours 
        price_limit: input_number.number_min_price_self_consumption
        first_hour: 6
        last_hour: 22
        starting_today: false
        number_of_hours: 4
        sequential: false
        failsafe_starting_hour: 7
        inversed: true
```
And input_number.number_min_price_self_consumption is just a input field where i can manually set my limit, currently 1,25, and the price is higher later today but it still didn't add it to the calendar for today, nor did it kick in the last few days.
I've also tried to set "price_limit: 1" directly in config and it didn't work. 

What am i doing wrong? 
Since inversed is set to true i expect it to kick in if the price is higher than the number i put in.

Again, it's probably me being a dumbass and missing something.

Thanks in advance

**Comment by kotope** [comment]
Hi!

price_limit needs to be decimal number. So if setting a constant number use 1.0 instead 1. 
Bad implementation from my part, sorry about that and I promise to fix it 'soon' :-)

For the dynamic entity id I'm not sure why it won't work. Do you get any errors in the logs when entering input_number ?

With the dynamic entity used the value is 'locked in' straight after new Nord Pool prices are received unless trigger_hour is used.
So when the Nord Pool data comes in your data will be locked until expiration happens. Expiration is straight after last_hour has passed.
So once you've received data you can't change it until expiration. Of course you can call the service to clear data and force recalculation in the middle of session. This will affect current schedule obviously.

**Reply by kawlsson** [reply]
Thanks for the very quick response! Impressive! :)
I'll give it a shot and go through the logs again.
Thanks again!

**Reply by kotope** [reply]
As I'm not using the dynamic entity value on price_limit on my 'production' there's a possibility of a bug as well obviously.
I'll try to find a slot today to see if it works as should :-)
